src/faqs/approaching-a-DSA-problem.md

```diff
@@ -0,0 +1,15 @@
+# How to approach a DSA problem ?
+#### Understanding the problem.
+The first step in approaching a DSA problem is to understand the essence of the problem itself. We must try and figure out what the problem demands. Not understanding the basic working of the problem is a big mistake. If the wording of the question is unclear, look at the testcases given for the particular problem and try to reverse-engineer the demands of the question.
+#### Clearing out a logical path.
+Now that the problem is understood, we must move on to the logic. We must have a clear picture of the answer's logic in our head. Once we have a clear "map" of this logic, we move on to implementing it in code. If the logic of your answer statement is fairly complicated, consider writing down comments while implementing the logic in your code. Once the logic is thought of, try to think of alternate ways to approach the problem if it doesn't meet the complexity constraints. Repeat this step until you're happy with your desired logic for the answer. Using pseudocode or diagrammatic representations to refine the algorithm before coding can help to understand the logic and essence of the program well.
+#### Effective Implementation. 
+Implement the logic in your code. As said before, consider adding comments to your code to make the logic "flow" better while solving it. Eventually, with enough problem solving, you won't require the comments to implement logic(s) anymore, though it is a good habit to do so. Another "cosmetic" tip is to name your variables well. Try to use as little variables as possible to minimize the storage space required for your code. Try to include various problem-solving paradigms such as greedy, dynamic programming, divide and conquer paradigms to guide the design of your algorithms.
+#### Testing and submission.
+Once implemented, test. Your code should be tested on a exhaustive set of inputs and outputs. This exhaustive search will allow you to find and fix any edge cases that slipped under the radar the first time. Try to think about why your code breaks at that edge case and see if it's a basic logical fallacy in your code, or simply a case that could have slipped between the cracks during the implementation. Try  using systematic debugging techniques such as print statements, debugging tools, or test case generators to identify and resolve errors more efficiently. This would allow you to pinpoint the source of all the errors and allow you to fix it surgically.
```

Comment:
```suggestion
- Once implemented, test. Your code should be tested on a exhaustive set of inputs and outputs. This exhaustive testing will allow you to find and fix any edge cases that slipped under the radar the first time. 
- Try to think about why your code breaks at that edge case and see if it's a basic logical fallacy in your code, or simply a case that could have slipped between the cracks during the implementation. 
- Try  using systematic debugging techniques such as print statements, debugging tools, or test case generators to identify and resolve errors more efficiently. This would allow you to pinpoint the source of all the errors and allow you to fix it surgically.
```